docs/DeveloperGuide.md

```diff
@@ -176,6 +177,87 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
+### Add student feature: `add -s`
+
+#### Implementation
+
+The add student feature adds a student contact to TutorAid. A student contact consists of details such as: the 
+student's name, student's contact number, the parent's name and parent's contact number. 
+
+The feature is mainly implemented by the following methods:
+* `AddStudentCommand#execute()` —Creates a student object and adds it to TutorAid.
+  
+It is also additionally facilitated by the methods below:
+* `TutorAidParser#parseCommand` —Checks for the command word that is required for the addition of a student.
+* `AddCommandParser#parse` —Checks for the command flag that specifies the addition of a student.
+* `AddStudentCommandParser#parse` —Parses the individual arguments to create a student object.
+
+Given below is an example of what happens when the user attempts to add a student to TutorAid by entering 
+a command `add -s sn/John Doe …​`:
+
+1. The command is first passed into `TutorAidParser#parseCommand`, which extracts the first keyword of every command. 
+   Since the keyword `add` would be extracted, the remaining arguments of the command (`-s sn/John Doe …​`) are passed 
+   then into `AddCommandParser#parse`.
+
+2. `AddCommandParser#parse` extracts the command flag `-s` at the start of its argument, which denotes the addition 
+   of a student contact. Thus, the remaining (`sn/John Doe …​`) is then passed into `AddStudentCommandParser#parse`.
+
+3. Each of the  different arguments of student contact, such as the student name, student contact number, parent name 
+   and parent number, is parsed by `AddStudentCommandParser#parse` based on the given input. 
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** 
+At this point, if `AddStudentCommandParser#parse` detects that no student name has been supplied, the command will fail 
+its execution and `ParseException` will be thrown.
+</div>
+
+4. For optional parameters, which are all parameters other the student's name, if the argument is not supplied by the 
+   user, a default argument (`""`) is supplied by the method. These individual arguments are then passed into 
+   `Model#Student()` to create a student object. 
+
+5. Lastly, the `AddStudentCommand#execute` is called upon to add the student into TutorAid, which returns a
+   `CommandResult` object to notify the user that the student has been successfully added. 
+   The student details are stored in `tutoraid.json`.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** 
+If the student object created appears to be a duplicate of an existing contact (all contact fields have the same), the 
+new student object will not be stored in TutorAid and user will be alerted of the duplicate instead. 
+</div>
+
+Below is the sequence diagram that depicts the `Logic` components involved in process:
+![AddStudent](images/AddStudentSequenceDiagram.png)
```

Comment:
For the sequence diagram, I believe the activation bar for the constructor of `:AddStudentCommand` should have a return arrow pointing back to the activation bar of `:AddStudentCommandParser`, and not have arrows extending from it? 
I also think that `addStudents(student)` is only called after `:LogicManager` calls the `execute` method of `:AddStudentCommand`?